src/boot/build.c

```diff
@@ -1,39 +1,12 @@
 #include "versions.h"
 
-#if PLATFORM_N64
-const char gBuildCreator[] = "zelda@srd44";
-#else
-const char gBuildCreator[] = "zelda@srd022j";
+#ifndef STRINGIFY
+# define STRINGIFY(s) #s
 #endif
-
-#if OOT_VERSION == NTSC_1_0
-const char gBuildDate[] = "98-10-21 04:56:31";
-#elif OOT_VERSION == NTSC_1_1
-const char gBuildDate[] = "98-10-26 10:58:45";
-#elif OOT_VERSION == PAL_1_0
-const char gBuildDate[] = "98-11-10 14:34:22";
-#elif OOT_VERSION == NTSC_1_2
-const char gBuildDate[] = "98-11-12 18:17:03";
-#elif OOT_VERSION == PAL_1_1
-const char gBuildDate[] = "98-11-18 17:36:49";
-#elif OOT_VERSION == GC_JP
-const char gBuildDate[] = "02-10-29 23:49:53";
-#elif OOT_VERSION == GC_JP_MQ
-const char gBuildDate[] = "02-10-30 00:15:15";
-#elif OOT_VERSION == GC_US
-const char gBuildDate[] = "02-12-19 13:28:09";
-#elif OOT_VERSION == GC_US_MQ
-const char gBuildDate[] = "02-12-19 14:05:42";
-#elif OOT_VERSION == GC_EU_MQ_DBG
-const char gBuildDate[] = "03-02-21 00:16:31";
-#elif OOT_VERSION == GC_EU
-const char gBuildDate[] = "03-02-21 20:12:23";
-#elif OOT_VERSION == GC_EU_MQ
-const char gBuildDate[] = "03-02-21 20:37:19";
-#elif OOT_VERSION == GC_JP_CE
-const char gBuildDate[] = "03-10-08 21:53:00";
-#else
-#error "Unsupported OOT_VERSION"
+#ifndef EXPAND_AND_STRINGIFY
+# define EXPAND_AND_STRINGIFY(s) STRINGIFY(s)
 #endif
 
+const char gBuildCreator[] = EXPAND_AND_STRINGIFY(BUILD_CREATOR);
+const char gBuildDate[] = EXPAND_AND_STRINGIFY(BUILD_DATE) " " EXPAND_AND_STRINGIFY(BUILD_TIME);
```

Comment:
I find it kinda weird that BUILD_DATE and BUILD_TIME are defined as sequences of C identifiers instead of as string literals. What about passing `-DBUILD_DATE="03-10-08"` instead, and skipping the stringification here? (It might need to be written as `-DBUILD_DATE='"03-10-08"'` or `-DBUILD_DATE=\"03-10-08\"` to get make/bash to cooperate, not sure)

Also, why split BUILD_DATE/BUILD_TIME in two instead of having only one define?

Reply:
Date and time are different since I had trouble getting it to work correctly otherwise, probably due to some make/bash issue. Also separate just makes more sense since it better mirrors the behavior of `__DATE__` and `__TIME__`, so I think it should stay this way.

I can try to do the stringification in the makefile though

Reply:
I see, yeah stringifying it in the makefile would make it even more like `__DATE__` and `__TIME__` then

Reply:
`CPP_DEFINES += -DBUILD_DATE=\"$(BUILD_DATE)\" -DBUILD_TIME=\"$(BUILD_TIME)\"` seems to work